app/models/notification.rb

```diff
@@ -12,10 +12,38 @@
 
 class Notification < ActiveRecord::Base
   belongs_to :user
+  after_save :send_gcm_notification
 
   scope :unread, -> { where read: false }
 
   def read!
     update_attributes read: true
   end
+
+  def create
+
+  end
+
+  def send_gcm_notification
+      if !Rpush::Gcm::App.find_by_name("tappb")
+        app = Rpush::Gcm::App.new
+        app.name = "tappb"
+        app.auth_key = Rails.application.secrets.fcm_secret
+        app.connections = 1
+        app.save!
+      end
+
+      n = Rpush::Gcm::Notification.new
+      n.app = Rpush::Gcm::App.find_by_name("tappb")
+      n.registration_ids = user.android_device_registration_tokens.all.map{|r| r.token}
+      n.data = { message: message }
```

Comment:
For clarity, you could use `@message` for instance variables.

Reply:
I've tried that, but it didn't seem to work for some reason (message was nil)

Reply:
Ah yes, because `message` is a method . `self.message` would be better then.